modules/data.atmosphere/R/check_met_input.R

```diff
@@ -0,0 +1,97 @@
+#' Check a meteorology data file for compliance with the PEcAn standard
+#'
+#' @param metfile Path of met file to check, as a scalar character.
+#' @param variable_table `data.frame` linking standard names to their
+#'   units. Defaults to [pecan_standard_met_table].
+#' @return `TRUE` (invisibly) if successful, otherwise throw an informative error.
+#' @author Alexey Shiklomanov
+#' @export
+check_met_input_file <- function(metfile, variable_table = pecan_standard_met_table) {
+
+  metfile <- normalizePath(metfile)
+
+  PEcAn.logger::severeifnot(
+    file.exists(metfile),
+    msg = glue::glue("File '{metfile}' does not exist.")
+  )
+
+  nc <- ncdf4::nc_open(metfile)
+
+  test_dims <- tryCatch(
+    testthat::test_that(
+      "Dimensions are correct.",
+      {
+        dimensions <- nc[["dim"]]
+        testthat::expect_type(dimensions, "list")
+        testthat::expect_equal(length(dimensions), 3)
+        testthat::expect_true("time" %in% names(dimensions))
+        testthat::expect_match(
+          ncdf4::ncatt_get(nc, "time", "units")[["value"]],
+          "^days since [[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}Z$"
+        )
+         testthat::expect_true("latitude" %in% names(dimensions))
+         testthat::expect_equal(
+           ncdf4::ncatt_get(nc, "latitude", "units")[["value"]],
+           "degrees_north"
+         )
+         testthat::expect_true("longitude" %in% names(dimensions))
+         testthat::expect_equal(
+           ncdf4::ncatt_get(nc, "longitude", "units")[["value"]],
+           "degrees_east"
```

Comment:
Another place where we can save typing if we trust the nc dimvals: `expect_equal(nc$dim$longitude$units, "degree_east")`

Reply:
See my above comment on `$` subsetting. This is actually a great example, because `nc...$unit` and `nc...$units` will both work, but `nc...[["unit"]]` will fail.

Reply:
I reluctantly concede your point :)